### PR TITLE
feat: CreateBossZones export and dependencies

### PR DESCRIPTION
**Pull request by mafewtm**
## Description

- Addition of an export for external initiation of the creation of the boss menu zones
- Addition of proper dependencies in the fxmanifest

## Checklist

- [x] I have personally loaded this code into an updated Qbox project and checked all of its functionality.
- [x] My code fits the style guidelines.
- [x] My PR fits the contribution guidelines.

### CODE REVIEW

**Review by solareon — APPROVED** [review]

**Review by dusan01 — APPROVED** [review]

**Comment by mafewtm** [comment]
Don't merge. Let Manason take a look first

File: fxmanifest.lua [hunk 1]
```diff
@@ -29,5 +29,11 @@ files {
     'config/client.lua',
 }
 
+dependency {
+    'qbx_core',
```
Comment (Manason) [inline fxmanifest.lua]: Don't make an fxmanifest dependency. Use ox_lib

File: fxmanifest.lua [hunk 2]
```diff
@@ -29,5 +29,11 @@ files {
...
+    'ox_lib',
```
Comment (Manason) [inline fxmanifest.lua]: ditto

File: fxmanifest.lua [hunk 3]
```diff
@@ -29,5 +29,11 @@ files {
...
+    'oxmysql',
```
Comment (Manason) [inline fxmanifest.lua]: ditto

File: client/main.lua [hunk 4]
```diff
@@ -148,58 +148,75 @@ function OpenBossMenu(groupType)
...
+---@param size? vector3 uses vec3(1.5, 1.5, 1.5) if not set
+---@param rotation? number uses 0.0 if not set
+local function createBossZones(groupName, groupType, coords, size, rotation)
+    size = size or vec3(1.5, 1.5, 1.5)
+    rotation = rotation or 0.0
+
+    createZones(groupName, groupType, coords, size, rotation)
+end
+
+exports('CreateBossZones', createBossZones)
```
Comment (Manason) [inline client/main.lua]: Shouldn't this be singular, not plural?

File: client/main.lua [hunk 5]
```diff
@@ -148,58 +148,75 @@ function OpenBossMenu(groupType)
...
 
+---Creates a boss zone for the specified group
+---@param groupName string Name of the group
+---@param groupType 'job'|'gang' Type of group
+---@param coords vector3 Coordinates of the zone
+---@param size? vector3 uses vec3(1.5, 1.5, 1.5) if not set
+---@param rotation? number uses 0.0 if not set
+local function createBossZones(groupName, groupType, coords, size, rotation)
+    size = size or vec3(1.5, 1.5, 1.5)
+    rotation = rotation or 0.0
```
Comment (Manason) [inline client/main.lua]: Should we move these defaults to be in createZones so that config can make use of them as well?

File: client/main.lua [hunk 6]
```diff
@@ -148,58 +148,75 @@ function OpenBossMenu(groupType)
...
     end
 end
 
+---Creates a boss zone for the specified group
+---@param groupName string Name of the group
+---@param groupType 'job'|'gang' Type of group
+---@param coords vector3 Coordinates of the zone
+---@param size? vector3 uses vec3(1.5, 1.5, 1.5) if not set
+---@param rotation? number uses 0.0 if not set
+local function createBossZones(groupName, groupType, coords, size, rotation)
```
Comment (Manason) [inline client/main.lua]: Consider using a table for the parameters. I think the table pattern provides a more readable function call and also allows us to more easily introduce changes in the future without breaking the API

**Review by Manason — CHANGES_REQUESTED** [review]

**Pull request merged**